File: src/main/kotlin/cn/har01d/notebook/vo/SystemStats.kt
package cn.har01d.notebook.vo

data class SystemStats(
        val user: UserStats,
        val note: NoteStats,
        val notebooks: Long,
        val categories: Long,
        val tags: Long,
)

data class NoteStats(
        val total: Long,
        val deleted: Long,
        val views: Long,
        val public: Long,
        val secret: Long,
        val private: Long,
)

data class UserStats(
        val total: Long,
        val online: Long,
)

File: web-ui/src/models/SystemStats.ts
export class NoteStats {
  total: number = 0
  views: number = 0
  deleted: number = 0
  public: number = 0
  secret: number = 0
  private: number = 0
}

export class UserStats {
  total: number = 0
  online: number = 0
}

export class SystemStats {
  user: UserStats = new UserStats()
  note: NoteStats = new NoteStats()
  notebooks: number = 0
  categories: number = 0
  tags: number = 0
}
